aptos-move/framework/supra-framework/sources/multisig_voting.move

```diff
@@ -0,0 +1,1219 @@
+            // If they have the key, we will remove it, because a single-step proposal that doesn't need this key.
+        } else if (simple_map::contains_key(&mut metadata, &is_multi_step_in_execution_key)) {
+            simple_map::remove(&mut metadata, &is_multi_step_in_execution_key);
+        };
+
+        table::add(&mut voting_forum.proposals, proposal_id, Proposal {
+            proposer,
+            creation_time_secs: timestamp::now_seconds(),
+            execution_content: option::some<ProposalType>(execution_content),
+            execution_hash,
+            metadata,
+            min_vote_threshold,
+            expiration_secs,
+            yes_votes: 0,
+            no_votes: 0,
+            is_resolved: false,
+            resolution_time_secs: 0,
+        });
+
+        if (std::features::module_event_migration_enabled()) {
+            event::emit(
+                CreateProposal {
+                    proposal_id,
+                    execution_hash,
+                    expiration_secs,
+                    metadata,
+                    min_vote_threshold,
+                },
+            );
+        };
+        event::emit_event<CreateProposalEvent>(
+            &mut voting_forum.events.create_proposal_events,
+            CreateProposalEvent {
+                proposal_id,
+                execution_hash,
+                expiration_secs,
+                metadata,
+                min_vote_threshold,
+            },
+        );
+
+        proposal_id
+    }
+
+    /// Vote on the given proposal.
+    ///
+    /// @param _proof Required so only the governance module that defines ProposalType can initiate voting.
+    ///               This guarantees that voting eligibility and voting power are controlled by the right governance.
+    /// @param voting_forum_address The address of the forum where the proposals are stored.
+    /// @param proposal_id The proposal id.
+    /// @param should_pass Whether the votes are for yes or no.
+    public fun vote<ProposalType: store>(
+        _proof: &ProposalType,
+        voting_forum_address: address,
+        proposal_id: u64,
+        should_pass: bool,
+    ) acquires VotingForum {
+        let voting_forum = borrow_global_mut<VotingForum<ProposalType>>(voting_forum_address);
+        let proposal = table::borrow_mut(&mut voting_forum.proposals, proposal_id);
+        // Voting might still be possible after the proposal has enough yes votes to be resolved early. This would only
+        // lead to possible proposal resolution failure if the resolve early threshold is not definitive (e.g. < 50% + 1
+        // of the total voting token's supply). In this case, more voting might actually still be desirable.
+        // Governance mechanisms built on this voting module can apply additional rules on when voting is closed as
+        // appropriate.
+        assert!(!is_voting_period_over(proposal), error::invalid_state(EPROPOSAL_VOTING_ALREADY_ENDED));
```

Comment:
Is this assertion relevant for multisig voting? Since we do not want proposal to become resolvable just because some time period has passed. Proposal should resolve only if `k` of `n` approval votes are received.

Reply:
Could you please confirm whether we need to set a voting `expiration_secs`?
if the required number of votes is not reached within the given expiration time, should the proposal be aborted?